test/connectors/AuthConnectorSpec.scala

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import config.WSHttp
+import connectors.AuthConnector.InvalidCredentials
+import model.{BearerToken, LoginDetails, Role}
+import org.joda.time.DateTime
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{BeforeAndAfterEach, Matchers}
+import play.api.libs.json.Json
+import uk.gov.hmrc.crypto.Protected
+import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+
+class AuthConnectorSpec extends UnitSpec with Matchers with ScalaFutures with WiremockSugar with BeforeAndAfterEach with WithFakeApplication {
+
+  override def beforeEach() {
+    startMockServer()
+  }
+
+  override def afterEach() {
+    stopMockServer()
+  }
+
+  trait Setup {
+    implicit val hc = HeaderCarrier()
+
+    val connector = new AuthConnector {
+      override val http = WSHttp
+      override val internalUserAuthBaseUrl: String = s"$wireMockUrl/auth/authenticate/user"
+    }
+
+    case class AuthBackendResponse(access_token: BearerToken, group: Option[String], roles: Option[Set[Role]])
+
+    implicit val format = Json.format[AuthBackendResponse]
+  }
+
+  "login" should {
+
+    "successfully authenticate user" in new Setup {
+      val loginDetails = LoginDetails("userName", Protected("password"))
+      val authenticateRequestJson = Json.toJson(loginDetails).toString()
+      val authenticateResponseJson = Json.toJson(
+        AuthBackendResponse(
+          BearerToken("bearer_token", DateTime.now().plusMinutes(10)),
+          Some("group"),
+          Some(Set(Role.APIGatekeeper))
+        )).toString()
+
+
+      stubFor(post(urlEqualTo("/auth/authenticate/user"))
+        .withRequestBody(equalToJson(authenticateRequestJson))
+        .willReturn(aResponse().withStatus(200)
+          .withHeader("Content-Type", "application/json")
+          .withBody(authenticateResponseJson)
+        ))
+
+      val result = await(connector.login(loginDetails))
+      verify(1, postRequestedFor(urlMatching("/auth/authenticate/user")).withRequestBody(equalToJson(authenticateRequestJson)))
+
+      result.roles.get shouldBe Set(Role.APIGatekeeper)
+      result.access_token.authToken shouldBe "bearer_token"
+    }
+
+    "return InvalidCredentials if authentication failed" in new Setup {
+      val loginDetails = LoginDetails("userName", Protected("password"))
+
+      stubFor(post(urlEqualTo("/auth/authenticate/user"))
+        .willReturn(aResponse().withStatus(401)))
+
+      val result = await(connector.login(loginDetails).failed)
+      result shouldBe a[InvalidCredentials]
```

Comment:
I've found the output a little bit more useful from intercept: 

```scala
intercept[InvalidCredentials](await(connector.login(loginDetails))
```

The reason is that connector.login(loginDetails).failed will throw if there's a success, which gives you an error like: 

```scala
@ await(Future.successful(10).failed)
java.util.NoSuchElementException: Future.failed not completed with a throwable.
	scala.concurrent.Future$$anonfun$failed$1.apply(Future.scala:192)
	scala.concurrent.Future$$anonfun$failed$1.apply(Future.scala:190)
        ...
```

Whereas intercept shows you something like "expected error InvalidCredentials wasn't thrown"